code-reviews/README.md

```diff
@@ -1,12 +1,46 @@
+# Code Reviews
+
 ## For the reviewer
 
-* Avoid judgement - assume that everyone is intelligent and well meaning.
-* Ask questions and get clarifications.  These may spark new ideas for the
+- Avoid judgement - assume that everyone is intelligent and well meaning.
+- Ask questions and get clarifications. These may spark new ideas for the
   developer that has opened the pull request and help you understand why a
   change is happening.
-
-[todo]
+- Find something positive to say about the work you're reviewing, especially if
+  you have some changes to request. People are more receptive to feedback if
+  you can frame the discussion positively.
 
 ## Having your code reviewed
 
-[todo]
+- Remember that the person reviewing your work does not have as much context as
+  you do. Provide them with as much context as you feel will be nessicary for
+  them to understand the changes you're proposing.
+- A great way to provide context to the reviewer is to use Github's review
+  features. For instance, if there is a line of code that you think will be
+  confusing, comment inline on code in your own PR so that the reviewer will
+  see it. You can find an [example of that in action](https://github.com/adaptdk/tufts-acsf/pull/335/files/fcfeee988c0082a24949cf18e66464fc518bc7e9#diff-8d31c81cc321d8731734e8da9e3af11a10306595cb94532b4d47c226582d9b32) on the Tufts ACSF proeject.
+- Always provide thorough testing steps so that the reviewer can test appropriately. Testing steps include:
+  - Any steps needed to get your environment into the state that should be tested
+  - Any manual setup setps that need to happen prior to testing
+  - Testing steps + expected outcomes
+
+## Work in small batches
```

Comment:
I'm curious what you think of this section @jesseday. The idea is taken from DevOps Handbook, which took the idea from Toyota's single piece flow manufacturing methodology (I'm pretty sure the same idea was in the Phoenix Project too).

It might be harder for devs to break up PRs like this, but my guess is that once they get past that learning curve it will make our overall process more efficient.

Reply:
I'm definitely on board with this.  I do think it would take some practice figuring out where the lines are and when a PR needs to be broken up, but I've reviewed (and probably opened) some large PRs that end up being hard to wrap your head around.  I'd like to try it this way.

The one thing that I will say is that we often have small tickets that offer reasonable lines of where a PR should be broken up, so I don't anticipate this being something that we run into a lot.